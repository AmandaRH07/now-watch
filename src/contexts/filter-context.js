import React, { createContext, useContext, useState } from 'react';

const FilterContext = createContext()
const FilterProvider = ({ children }) => {
  const [checkboxService, setCheckboxService] = useState({
    apple: false,
    prime: false,
    disney: false,
    hbo: false,
    netflix: false,
    starz: false
  });

  const [checkboxType, setCheckboxType] = useState({
    movie: false,
    series: false,
  });

  const [checkboxGenre, setCheckboxGenre] = useState({
    Acao28: false,
    Animacao16: false,
    Biografia1: false,
    Comedia38: false,
    Documentario99: false,
    Drama18: false,
    FiccaoCientifica878: false,
    Romance10749: false,
    Terror27: false
  });

  const [filterService, setFilterService] = useState([]);
  const [filterType, setFilterType] = useState([]);
  const [filterGenre, setFilterGenre] = useState([]);
  const [responseHasMore, setResponseHasMore] = useState();
  const [responseNextCursor, setResponseNextCursor] = useState();

  const [responseData, setResponseData] = useState([]);
  return (
    <FilterContext.Provider
      value={{
        checkboxService,
        setCheckboxService,
        checkboxType,
        setCheckboxType,
        checkboxGenre,
        setCheckboxGenre,
        filterService,
        setFilterService,
        filterType,
        setFilterType,
        filterGenre,
        setFilterGenre,
        responseHasMore,
        setResponseHasMore,
        responseNextCursor,
        setResponseNextCursor,
        responseData,
        setResponseData
      }}>
      {children}
    </FilterContext.Provider>
  )
}

export const useFilterContext = () => useContext(FilterContext);
export { FilterProvider, FilterContext }