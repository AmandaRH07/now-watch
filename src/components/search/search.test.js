import { BrowserRouter } from "react-router-dom"
import SearchComponent from "."
import { render } from "@testing-library/react"
import { FilterProvider } from "../../contexts/filter-context"

const renderComponent = () => (
  render(
    <BrowserRouter>
      <FilterProvider>
        <SearchComponent />
      </FilterProvider>
    </BrowserRouter>
  )
)

describe('<SearchComponent/>', () => {

  it('Should render SearchComponent', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  })
})