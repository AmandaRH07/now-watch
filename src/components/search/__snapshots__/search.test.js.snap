// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchComponent/> Should render SearchComponent 1`] = `
<div>
  <div
    class="sc-aXZVg ggAGdP"
  >
    <div
      class="sc-gEvEer eAFNFZ"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-eqUAAy gxAfTl css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard css-z5e0z9-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="input-with-sx"
          id="input-with-sx-label"
        >
          Search
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-v4u5dn-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-nz481w-MuiInputBase-input-MuiInput-input"
            id="input-with-sx"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-fqkvVR byzlMI css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
